### PR TITLE
test(encoders): add tests

### PR DESCRIPTION
**Pull request by Yongxuanzhang**

### CODE REVIEW

File: tests/unit/executors/encoders/test_numeric.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+def test_test_transformencoder(model):
+    train_data = np.random.rand(100, input_dim)
+    encoder = TransformEncoder(output_dim=target_output_dim)
+    encoder.model = model
+    encoder.train(train_data)
+
+    test_data = np.random.rand(10, input_dim)
+    encoded_data = encoder.encode(test_data)
+    assert encoded_data.shape == (test_data.shape[0], target_output_dim)
+    assert type(encoded_data) == np.ndarray
```
Comment (bwanglzu) [inline tests/unit/executors/encoders/test_numeric.py:33]: i would do assert isinstance(encoded_data, np.ndarray)

**Review by bwanglzu — COMMENTED** [review]
left a comment above

File: tests/unit/executors/encoders/test_numeric.py (hunk 1 above)
Comment (Yongxuanzhang) [reply tests/unit/executors/encoders/test_numeric.py:33]: encode() will use the trained model. Can we do that in separate tests?

Comment (Yongxuanzhang) [reply tests/unit/executors/encoders/test_numeric.py:33]: It makes sense to cover more. I will try it

File: tests/unit/executors/encoders/test_numeric.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+def test_transformencoder_train():
+    train_data = np.random.rand(100, input_dim)
+    encoder.model = simple_model
+    encoder.train(train_data)
+    assert encoder.train
+
+
+def test_transformencoder_test():
```
Comment (JoanFM) [inline tests/unit/executors/encoders/test_numeric.py:29]: test_transformer_encode?

File: tests/unit/executors/encoders/test_numeric.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+simple_model = SimpleModel()
+encoder = TransformEncoder(output_dim=target_output_dim)
+
+
+def test_transformencoder_train():
+    train_data = np.random.rand(100, input_dim)
+    encoder.model = simple_model
+    encoder.train(train_data)
+    assert encoder.train
```
Comment (JoanFM) [inline tests/unit/executors/encoders/test_numeric.py:26]: what is exactly being asserted here?

Comment (Yongxuanzhang) [reply tests/unit/executors/encoders/test_numeric.py:26]: A typo. will be fixed in the next commit. Intended to `assert encoder.is_trained`

File: tests/unit/executors/encoders/test_numeric.py (hunk 2 above)
Comment (Yongxuanzhang) [reply tests/unit/executors/encoders/test_numeric.py:29]: Thanks!

**Comment by Yongxuanzhang** [comment]
Sorry, I forgot to request reviews again. @bwanglzu @JoanFM

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**